src/chains-config/index.ts

```diff
@@ -0,0 +1,51 @@
+import { ApiPromise } from '@polkadot/api';
+import AbstractController from 'src/controllers/AbstractController';
+import { AbstractService } from 'src/services/AbstractService';
+
+import { controllers } from '../controllers';
+import { ControllerConfig } from '../types/chains-config';
+import { defaultControllers } from './defaultControllers';
+import { kulupuControllers } from './kulupuControllers';
+import { mandalaControllers } from './mandalaControllers';
+/**
+ * Return an array of instantiated controller instances based off of a `specName`.
+ *
+ * @param api ApiPromise to inject into controllers
+ * @param implName
+ */
+export function getControllersForSpec(
+	api: ApiPromise,
+	specName: string
+): AbstractController<AbstractService>[] {
+	switch (specName) {
+		case 'kulupu':
+			return getControllersFromConfig(api, kulupuControllers);
+		case 'mandala':
+			return getControllersFromConfig(api, mandalaControllers);
+		default:
+			return getControllersFromConfig(api, defaultControllers);
+	}
+}
+
+/**
+ * Return an array of instantiated controller instances based off of a
+ * `ControllerConfig`.
+ *
+ * @param api ApiPromise to inject into controllers
+ * @param config controller mount configuration object
+ */
+function getControllersFromConfig(api: ApiPromise, config: ControllerConfig) {
+	// If we don't typecast here, tsc thinks its just [string, any][]
+	const controllersToInclude = Object.entries(config) as [
+		keyof typeof controllers,
+		boolean
+	][];
+
+	return controllersToInclude.reduce((acc, [controllerName, shouldMount]) => {
+		if (shouldMount) {
+			return acc.concat(new controllers[controllerName](api));
+		}
+
+		return acc;
```

Comment:
```suggestion
		if (shouldMount) {
			acc.push(new controllers[controllerName](api));
		}

		return acc;
```

Doesn't `concat()` create a new array?

Reply:
This is a good point... not sure why but my brain was telling me to make `acc` immutable.